src/layerstack.stanza

```diff
@@ -576,49 +579,106 @@ doc:\<DOC>
        copper-17_5um
        core
        copper-17_5um
-       prepreg
+       prepreg2         ; two layers of prepreg2 in the center, not one single core/prepreg layer 
+       prepreg2         ;
+       copper-17_5um
+       core
+       copper-17_5um
        prepreg
        copper-35um
      ]
+    ```
 
   @param name The name of the LayerStack
-  @param top-layers The pairs of mechanical and dielectric layers, from outer to inner layers.
-  @param description The description of the LayerStack
+  @param top-layers A tuple of pairs of the mechanical layer and one dielectric layer or 
+    of the mechanical layer and a tuple of two or more dielectric layers, from outer to inner layers.
+  @param description (optional) The description of the LayerStack
+  @param soldermask (optional) The soldermask layer to be added as the outermost layer
+  @param even-layers? (optional) Whether there are an even number of layers. The default is `false`.
 <DOC>
-public defn make-layer-stack (name:String, outers:Tuple<[LayerSpec, LayerSpec]|LayerSpec>
-      -- description:String = ?, soldermask:LayerSpec = ?) -> LayerStack :
+public defn make-layer-stack (name:String,
+  outers:Tuple<[LayerSpec LayerSpec]|[LayerSpec Tuple<LayerSpec>]>
+      -- description:String = ?, 
+         soldermask:LayerSpec = ?,
+         even-layers?:True|False = false) -> LayerStack :
   val stack = #LayerStack(One(name), description)
   ;Access pairs of outer-layers in reverse order
   for outer in in-reverse(outers) do:
-    match(outer):
-      (outer:LayerSpec):
-        ; add one prepreg layer to top and bottom
-        add-top(outer, stack)
-        add-bottom(outer, stack)
-      ([metal, insul]:[LayerSpec LayerSpec]):
-        ; add [copper, prepreg] layers to top and bottom
+    ;Verify input
+    if outer[1] is Tuple<LayerSpec> and empty?(outer[1] as Tuple<LayerSpec>) :
+      throw(Exception("Invalid argument: The tuple of dialectric layers is empty. Expect [LayerSpec Tuple<LayerSpec>]. Found %_" % [outer]))
+    if not even-layers? :
+      if outer[1] is LayerSpec :
+        val [metal insul] = outer as [LayerSpec LayerSpec]
         add-symmetric(metal, insul, stack)
+      else : ; outer[1] is Tuple<LayerSpec>
+        val [metal insuls] = outer as [LayerSpec Tuple<LayerSpec>]
+        ; add the innermost layers first
+        val n-insuls = length(insuls) ; insuls[n-insuls - 1] is the innermost layer
+        if n-insuls == 1 : ; [copper [dielectric]] => treat it as [copper dielectric]
+          add-symmetric(metal, insuls[0], stack)
+        else :
+          add-symmetric(insuls[n-insuls - 2], insuls[n-insuls - 1], stack)
+          ; then add the rest of the layers
+          val layers = to-list $ cat([metal], insuls[0 to n-insuls - 2]) ; from outer to inner layers
+          add-top(reverse(layers), stack) ; ordered from inner to outer
+          add-bottom(reverse(layers), stack)
+    else : ; even-layers? = true
+      if outer[1] is LayerSpec :
```

Comment:
Can we please use a `match` statement instead of using `if ... is` ? 

The `if ... is` is an anti-pattern in stanza and the idiomatic mechanism is the `match` clause.